### DISCUSSION TITLE
DeckGL with ReactMapGL Marker

### DISCUSSION THREAD

**Discussion by vacaramin** [visgl/deck.gl#8943] [General]
Is it possible to make it so that ReactMapGL's [Marker](http://visgl.github.io/react-map-gl/docs/api-reference/marker) go on top of the DeckGL layers? The previous fixes do not work in the new versions of deckGL and React Map GL.

**Comment by vacaramin** [comment]
Anyone on this issue, looking for a fix in the latest version I found it, use a mapbox-overlay for the additional layers you add. So you probably are stuck, with having a deckGL parent and within it a mapbox. and somehow the mapbox layer comes on top. and you can't use the markers or anything that mapbox renders in that case, just remove the deckGl parent it a overlay child to the mapboxgl, this article will help you out a lot

[Life saver link](https://deck.gl/docs/api-reference/mapbox/mapbox-overlay)

and some snippets that'll help you
```
function DeckGLOverlay(props: DeckProps) {
  const overlay = useControl<MapboxOverlay>(() => new MapboxOverlay(props));
  overlay.setProps(props);
  return null;
}
```
```
const layers = [
      new [Your layer type]({
        id: "trips",
        data: DATA_URL.TRIPS,
        getPath: (d) => d.path,
        // additional attributes of your layers. 
      }),
    ];
```
```
<DeckGLOverlay layers={layers} />
```

**Reply by colindm** [reply]
thanks you so much this saved me hours
